src/Centreon/Application/Normalizer/IconUrlNormalizer.php

```diff
@@ -34,43 +33,23 @@ class IconUrlNormalizer implements ContextAwareNormalizerInterface
     private const IMG_DIR = '/img/media';
 
     /**
-     * @inheritDoc
+     * Concat base url with icon path to get full url
+     * {@inheritDoc}
      */
-    public function normalize($resource, $format = null, array $context = [])
+    public function normalize($icon, $format = null, array $context = [])
     {
-        // normalize resource icon
-        if ($resource->getIcon() !== null) {
-            $this->normalizeIcon($resource->getIcon());
-        }
-
-        // normalize parent resource icon
-        if ($resource->getParent() !== null && $resource->getParent()->getIcon() !== null) {
-            $this->normalizeIcon($resource->getParent()->getIcon());
+        if (isset($_SERVER['REQUEST_URI']) && preg_match('/^(.+)\/api\/.+/', $_SERVER['REQUEST_URI'], $matches)) {
+            $icon->setUrl($matches[1] . self::IMG_DIR . '/' . $icon->getUrl());
```

Comment:
shouldn't you check that matches[1] is found ? eg: REQUEST_URI does not contains the word "api"

Reply:
It cannot enter in this condition if $matches[1] does not exist ;)
Cause preg_match returns false if nothing is matched